mm2src/coins/z_coin.rs

```diff
@@ -376,6 +377,87 @@ impl ZCoin {
         }
     }
 
+    async fn spendable_notes_required_for_tx(
+        &self,
+        total_required: &BigDecimal,
+    ) -> MmResult<Vec<SpendableNote>, GenTxError> {
+        let spendable_notes = self
+            .spendable_notes_ordered()
+            .await
+            .map_err(|err| GenTxError::SpendableNotesError(err.to_string()))?;
+        let changes = self.z_fields.change_tracker.change_notes_iter();
+        let my_z_balance = self
+            .my_balance()
+            .compat()
+            .await
+            .mm_err(|err| GenTxError::Internal(err.to_string()))?;
+
+        if changes.is_empty() {
+            Ok(spendable_notes)
+        } else {
```

Comment:
The else block is way too large. Can you please do

```rs
if changes.is_empty() {
   return Ok(spendable_notes);
}
```

and release the else block to improve readability?

Reply:
outdated